tests/test_cnes_sonarqube.py

```diff
@@ -0,0 +1,251 @@
+            max_map_count = int(re.findall('vm.max_map_count = [0-9]+', str(proc_stdout))[0].split(' = ')[1])
+            # Hint: if this test fails, run: sudo sysctl -w vm.max_map_count=262144
+            assert max_map_count >= 262144
+        docker_client = docker.from_env()
+        lequalsonarqube_container_name="lequalsonarqube-compose"
+        # Use the compose file with an external database
+        print("Starting the service (sonarqube and postgres)...")
+        subprocess.run("docker-compose up -d", shell=True, check=True, capture_output=True)
+        # Wait for the SonarQube container to be configured
+        self.wait_cnes_sonarqube_ready(lequalsonarqube_container_name)
+        # Restart the SonarQube server but not the database
+        print("Restarting SonarQube server...")
+        docker_client.containers.get(lequalsonarqube_container_name).restart()
+        time.sleep(30)
+        self.wait_cnes_sonarqube_ready(lequalsonarqube_container_name, tail=10)
+        # Check SonarQube logs
+        config_logs = docker_client.containers.get(lequalsonarqube_container_name).logs()
+        subprocess.run("docker-compose down", shell=True, check=True, capture_output=True)
+        docker_client.volumes.get("tests_test_volume_compose_sonarqube_data").remove()
+        docker_client.volumes.get("tests_test_volume_compose_sonarqube_extensions").remove()
+        docker_client.volumes.get("tests_test_volume_compose_sonarqube_logs").remove()
+        docker_client.volumes.get("tests_test_volume_compose_postgresql").remove()
+        docker_client.volumes.get("tests_test_volume_compose_postgresql_data").remove()
+        # Hint: if this test fails, the server may have been reconfigured when not needed or the test did not wait long enough for the expected message to be logged
+        assert b"[INFO] CNES SonarQube: The database has already been filled with CNES configuration. Not adding anything." in config_logs
+
+    def test_no_password_no_run(self):
+        """
+        As a SonarQube user, I want the container not to start when I forget to
+        set the admin password so that the default admin password cannot be used.
+        """
+        docker_client = docker.from_env()
+        for password in ("", "admin"):
+            params = {"name": "tmp", "detach": True}
+            if password:
+                params['environment'] = {"SONARQUBE_ADMIN_PASSWORD": password}
+            docker_client.containers.run("lequal/sonarqube:latest", **params)
+            time.sleep(3)
+            output = docker_client.containers.get("tmp").logs()
+            docker_client.containers.get("tmp").remove(force=True)
+            # Hint: if this test fails, the server may have started with the default admin password or the test did not wait long enough for the expected message to be logged
+            assert b"Failed to start CNES SonarQube." in output
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)